doc/feature/spikeReport.md

```diff
@@ -0,0 +1,450 @@
+  it should be easy to write client code that works the same for both file
+  based and stream based reports without special conditions for stream based
+  reports.
+
+* For stream reports the user will be able to:
+ * Retrieve all the data until the end of stream in a simple loop without doing
+   any active wait (blocking calls are allowed).
+ * Know if the end of stream has been reached.
+ * Get all the data received so far without locking.
+ * Get information about the latest timestamp reached by the source in the
+   sender side even when there are no spikes at all (currently if the spike
+   report is completely empty it's not possible to make any progress until the
+   producer reaches the end of the report).
+
+* When data for a time interval is requested, the time intervals are always
+  closed on the left and open on the right.
+
+## API
+
+The classes presented below are not thread-safe except for const access.
+
+### Spikes container
+
+    Spikes
+    {
+    public:
+        Spikes();
+        ~Spikes();
+        /*
+         * Provided to allow Boost.Python to use return by value policies.
+         * Constant complexity.
+         */
+        Spikes( const Spikes& rhs );
+        /*
+         * Provided to allow Boost.Python to use return by value policies.
+         * Constant complexity.
+         */
+        Spikes& operator=( const Spikes& rhs );
+
+        class const_iterator; // implementation defined
+
+        const_iterator begin() const;
+        const_iterator end() const;
+        bool empty() const;
+        size_t size() const;
+        /* Get the spike at the nth position. No range check is performed */
+        const std::pair< float, uint32_t >& operator[]( size_t index ) const;
+
+        /*
+         * Get the start of the time window in milliseconds.
+         *
+         * UNDEFINED_TIMESTAMP if the container cannot be referred to any time
+         * window and it is empty. Otherwise, a time smaller or equal than the
+         * smallest spike time in the container.
+         *
+         * @see SpikesReport::getSpikes.
+         */
+        float getStartTime() const;
+
+        /*
+         * Get the end of the time window in milliseconds.
+         *
+         * UNDEFINED_TIMESTAMP if the container cannot be referred to any time
+         * window and it is empty. Otherwise, a time greater than the highest
+         * spike time in the container.
+         */
+        float getEndTime() const;
+    }
+
+
+With this API, it's relatively easy to provide a Boost.Python wrapping of the
+container inside a numpy array.
+
+### SpikeReport
+
+    SpikeReport : public boost::noncopyable
+    {
+    public:
+        SpikeReport( const URI& uri, int mode );
+
+        /*
+         * Open a report in read mode with a subset selections.
+         *
+         * @param uri
+         * @param subset The set of gids to be reported.
+         */
+        SpikeReport( const URI& uri, uint32_ts subset );
```

Comment:
I forgot about GIDSet. Yes, it should be that type.

For missing GIDs I think the right solution is to ignore them without any notice. Otherwise we need to know which are the valid GIDs at source, which is not unfeasible, but requires either loading all the data or changing the format for files and becomes quite complicated for streams.